app/src/main/java/com/randomappsinc/studentpicker/home/HomeActivity.java

```diff
@@ -208,20 +210,27 @@ protected void onResume() {
     public void onActivityResult(int requestCode, int resultCode, Intent data) {
         super.onActivityResult(requestCode, resultCode, data);
         if (requestCode == IMPORT_FILE_REQUEST_CODE) {
-            if (resultCode == Activity.RESULT_OK && data != null && data.getData() != null) {
-                Uri uri = data.getData();
-
-                // Persist ability to read from this file
-                int takeFlags = data.getFlags()
-                        & (Intent.FLAG_GRANT_READ_URI_PERMISSION
-                        | Intent.FLAG_GRANT_WRITE_URI_PERMISSION);
-                getContentResolver().takePersistableUriPermission(uri, takeFlags);
-
-                String uriString = uri.toString();
-                Intent intent = new Intent(this, ImportFromTextFileActivity.class);
-                intent.putExtra(Constants.FILE_URI_KEY, uriString);
-                startActivity(intent);
-            }
+            maybeOpenImportFileActivity(resultCode,data,true);
+        } else if (requestCode == IMPORT_CSV_REQUEST_CODE) {
+            maybeOpenImportFileActivity(resultCode,data,false);
+        }
+    }
+
+    private void maybeOpenImportFileActivity(int resultCode, Intent data, boolean isTextFile) {
```

Comment:
Even though there are only 2 cases, equating (not .txt) with (is .csv) is hard to read and won't scale if we support more import types. Let's create an [IntDef](https://developer.android.com/reference/android/support/annotation/IntDef) for the 2 file types. We can name it FileImportType and put it either in common or importdata. We will need to change the method signature of this method alongside the data passed between activities via intent.